crates/sui-framework/sources/event.move

```diff
@@ -2,6 +2,22 @@
 // SPDX-License-Identifier: Apache-2.0
 
 module sui::event {
+    /// An attachable event. Should be emitted once unpacked from a
+    /// a wrapper. Holds any custom event data inside.
+    struct Event<T: copy + drop + store> has store {
+        data: T
+    }
+
+    /// Create a new `Event` wrapper for custom `data`.
+    public fun create<T: copy + drop + store>(data: T): Event<T> {
+        Event { data }
+    }
+
+    /// Unpack contents of `Event` and `emit` them.
+    public fun emit_event<T: copy + drop + store>(evt: Event<T>) {
```

Comment:
we would need better naming here, bit strange to have `emit` and `emit_event`

Reply:
Absolutely agree! Open to suggestions

Reply:
Me too :P I came up with these names after all. But I hate them lol